### PR TITLE
adds dns support for BuddyDNS

### PR DESCRIPTION
**Pull request by piratimir**

### CODE REVIEW

**Comment by piratimir** [comment]
@tonybaloney check it man :)

File: libcloud/dns/drivers/buddyns.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+import sys
```
Comment (tonybaloney) [inline libcloud/dns/drivers/buddyns.py:19]: You definitely need the Apache license header in this file

File: libcloud/common/buddyns.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+from libcloud.common.base import ConnectionKey, JsonResponse
```
Comment (tonybaloney) [inline libcloud/common/buddyns.py:16]: And the header needs to go here

**Comment by tonybaloney** [comment]
@jetbird another great contribution! just a couple of small changes, you need to add the Apache license header to the 2 files I noted.

**Comment by piratimir** [comment]
@tonybaloney thanks man! Done :)

**Comment by Kami** [comment]
@jetbird Since this is a bigger change please make sure to submit an ICLA (http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) and select to notify the Libcloud project :)

File: libcloud/dns/drivers/buddyns.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+        :param ttl: TTL for new records (This is used through the extra param)
+        :type ttl: ``int``
+
+        :param extra: Extra attributes that are specific to the driver
+        such as ttl.
+        :type extra: ``dict``
+
+        :rtype: :class:`Zone`
+        Do not forget to pass the master in extra,
```
Comment (Kami) [inline libcloud/dns/drivers/buddyns.py:94]: I assume `master` key is only required for slave zones? or?

If so, you could probably throw if type is slave and 'master' not in extra or similar...

**Comment by Kami** [comment]
Is this pull request sill work in progress or considered complete?

I'm wondering because I only see methods for zone (domain) management, but no methods for managing records.

**Comment by Kami** [comment]
As Anthony has said, nice work so far :+1:

**Comment by piratimir** [comment]
@Kami the provider does not support operation on records.

File: libcloud/dns/drivers/buddyns.py (hunk 3 above)
Comment (piratimir) [inline libcloud/dns/drivers/buddyns.py:94]: According to the docs the key master is always required when doing a POST request for creating a new zone.
https://www.buddyns.com/support/api/v2/

**Comment by Kami** [comment]
ACK the receival of ICLA. Thanks.

**Comment by tonybaloney** [comment]
@jetbird one more thing, I tagged you in a commit since I missed it on the other 2 PRs.
You need to update libcloud/dns/providers.py

**Comment by Kami** [comment]
@jetbird I also noticed you don't have your git name and email fully configured.

Please follow the instructions at https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup and configure your git name and email. This way it's easier for us to track commits back to actual contributors.

**Pull request closed without merging**